### ISSUE TITLE
Provide an API to show hover-like tooltip

### ISSUE THREAD

**Issue by AlexeyRaga** [Microsoft/vscode#32088]
Please provide an API to trigger a hover-like tooltip.

Use case:
In my language extension, I want to let users know the type of the expression by pressing a combination of keys. This is similar to what hover does but:
- It is triggered by pressing a shortcut
- When the same shortcut is pressed again I want to widen the scope of the expression and show the type of a "bigger" expression, etc.

Being able to show custom tooltips is also useful in different scenarios for providing extra information about the code.

- VSCode Version: 1.14.2
- OS Version: MacOS

**Comment by jrieken** [comment]
@alexandrudima Do we already have an internal command that shows the hover?

**Comment by alexandrudima** [comment]
`editor.action.showHover` will show the hover at the current editor position.

**Comment by monk-time** [comment]
@alexandrudima, there are multiple issues with assigning a shortcut to `editor.action.showHover`:
- the hover appears with the same delay as when using a mouse, which is unnecessary when using a keyboard
- the hover blocks all other keyboard input and can be closed only with Esc, which is very disruptive

If the hover is shown using the shortcut, IMO it should not block anything, and it should be displayed for as long as it is valid, i.e. the cursor is on the same symbol. And it should still be possible to dismiss it with Esc or the same shortcut (basically making it a toggle).

There are two ways for how to handle the moment the cursor leaves the symbol that the hover is describing:
- remove the hover
- or keep showing the hover above the cursor, changing its content as the cursor moves over other symbols

Personally I'd prefer the latter, but having a pref would be even better.

**Comment by AlexeyRaga** [comment]
I totally agree with @monk-time and the suggested solution (keep showing) fits my use case perfectly.

**Comment by farzher** [comment]
also, it's currently impossible to show the alt hover with keyboard... need a solution for that.
(when you hold alt while hovering it shows different information).
here's the unsolved stackoverflow you find when googling for how to do this.

**Comment by mplanchard** [comment]
The main problem I have is that while there's an overridable keyboard shortcut for "show hover," there isn't a corresponding "hide hover." I have `Ctrl+[` mapped to `Esc` functionality pretty much everywhere, but I'm not able to dismiss a hover without pressing `Esc`. 

I'd also be okay with what @JeanMertz mentioned, and having the hover automatically disappear whenever typing or other actions are detected.

**Comment by eamodio** [comment]
It looks like many issues with the `editor.action.showHover` command have been resolved -- it seems to show up immediately, and it doesn't steal focus. It also get dismissed on almost any keystroke. Can this be closed or is there outstanding issues that still need to be addressed?

**Comment by ice1000** [comment]
🤔
